### ISSUE TITLE
Hybrid Fastconformer converging in training but not validation

### ISSUE THREAD

**Issue by MahmoudAshraf97** [NVIDIA-NeMo/NeMo#15264]
**Describe the bug**

I'm training a hybrid fastconformer medium with the following config:
<details><summary>Config</summary>
<p>

```yaml
model:
  tokenizer:
    dir: /home/mahmo/development/artifacts/ud_tokenizer_spe_bpe_v1024_max_4:v0
    type: bpe
  train_ds:
    augmentor:
      gain:
        prob: 0.5
        min_gain_dbfs: -10
        max_gain_dbfs: 10
      speed:
        prob: 0.5
        sr: 16000
        resample_type: kaiser_best
        min_speed_rate: 0.9
        max_speed_rate: 1.1
        num_rates: 5
      time_stretch:
        prob: 0.5
        min_speed_rate: 0.9
        max_speed_rate: 1.1
        num_rates: 5
        n_fft: 512
      white_noise:
        prob: 0.5
        min_level: -50
        max_level: -10
    manifest_filepath: /home/mahmo/development/data/tarred_everything/sharded_manifests_ud/manifest_{0..511}.json
    tarred_audio_filepaths: /home/mahmo/development/data/tarred_everything/audio_{0..511}.tar
    sample_rate: 16000
    batch_size: 32
    shuffle: true
    num_workers: 8
    pin_memory: true
    max_duration: 60
    min_duration: 0.1
    is_tarred: true
    use_lhotse: true
  sample_rate: 16000
  compute_eval_loss: false
  log_prediction: false
  skip_nan_grad: false
  model_defaults:
    enc_hidden: 256
    pred_hidden: 640
    joint_hidden: 640
  validation_ds:
    manifest_filepath: /home/mahmo/development/data/FsCfCtcLg-SpeUni1024-DI-TAWSEEM-EG-CA/val_FsCfCtcLg-SpeUni1024-DI-TAWSEEM-EG-CA_manifest_UD.json
    sample_rate: 16000
    batch_size: 64
    shuffle: false
    use_start_end_token: false
    num_workers: 8
    pin_memory: true
    max_duration: 32.0
    min_duration: 0.1
    use_dali: false
  preprocessor:
    _target_: nemo.collections.asr.modules.AudioToMelSpectrogramPreprocessor
    sample_rate: 16000
    normalize: per_feature
    window_size: 0.025
    window_stride: 0.01
    window: hann
    features: 80
    n_fft: 512
    frame_splicing: 1
    dither: 1.0e-05
    pad_to: 0
  spec_augment:
    _target_: nemo.collections.asr.modules.SpectrogramAugmentation
    freq_masks: 2
    time_masks: 10
    freq_width: 27
    time_width: 0.05
  encoder:
    _target_: nemo.collections.asr.modules.ConformerEncoder
    feat_in: 80
    feat_out: -1
    n_layers: 16
    d_model: 256
    use_bias: true
    subsampling: dw_striding
    subsampling_factor: 8
    subsampling_conv_channels: 256
    causal_downsampling: True
    ff_expansion_factor: 4
    self_attention_model: rel_pos
    n_heads: 4
    xscaling: true
    pos_emb_max_len: 5000
    use_pytorch_sdpa: false
    use_pytorch_sdpa_backends: []
    conv_kernel_size: 9
    conv_norm_type: layer_norm
    conv_context_size: 'causal'
    att_context_size: [[20, 0], [20, 1], [20,4], [20, 9]]
    att_context_style: chunked_limited # regular or chunked_limited
    dropout: 0.1
    dropout_pre_encoder: 0.1
    dropout_emb: 0.0
    dropout_att: 0.1
    stochastic_depth_drop_prob: 0.0
    stochastic_depth_mode: linear
    stochastic_depth_start_layer: 1
  use_cer: false
  ctc_reduction: mean_batch
  decoder:
    _target_: nemo.collections.asr.modules.RNNTDecoder
    normalization_mode: null # Currently only null is supported for export.
    random_state_sampling: false # Random state sampling: https://arxiv.org/pdf/1910.11455.pdf
    blank_as_pad: true # This flag must be set in order to support exporting of RNNT models + efficient inference.
    # num_classes: 1024

    prednet:
      pred_hidden: ${model.model_defaults.pred_hidden}
      pred_rnn_layers: 1
      t_max: null
      dropout: 0.2

  joint:
    _target_: nemo.collections.asr.modules.RNNTJoint
    log_softmax: null  # 'null' would set it automatically according to CPU/GPU device
    preserve_memory: false  # dramatically slows down training, but might preserve some memory
    fuse_loss_wer: false
    fused_batch_size: 32

    jointnet:
      joint_hidden: ${model.model_defaults.joint_hidden}
      activation: "relu"
      dropout: 0.2

  decoding:
    strategy: "greedy_batch" # can be greedy, greedy_batch, beam, tsd, alsd.

    greedy:
      max_symbols: 10
      use_cuda_graph_decoder: false

    beam:
      beam_size: 2
      return_best_hypothesis: False
      score_norm: true
      tsd_max_sym_exp: 50  # for Time Synchronous Decoding
      alsd_max_target_len: 2.0  # for Alignment-Length Synchronous Decoding

  aux_ctc:
    ctc_loss_weight: 0.3 # the weight used to combine the CTC loss with the RNNT loss
    use_cer: false
    ctc_reduction: 'mean_batch'
    decoder:
      _target_: nemo.collections.asr.modules.ConvASRDecoder
      feat_in: null
      num_classes: -1
      vocabulary: []
    decoding:
      strategy: "greedy_batch"
  interctc:
    loss_weights: []
    apply_at_layers: []
  loss:
    loss_name: "default"
    warprnnt_numba_kwargs:
      # FastEmit regularization: https://arxiv.org/abs/2010.11148
      # You may enable FastEmit to increase the accuracy and reduce the latency of the model for streaming
      # You may set it to lower values like 1e-3 for models with larger right context
      fastemit_lambda: 1e-2  # Recommended values to be in range [1e-4, 1e-2], 0.001 is a good start.
      clamp: -1.0  # if > 0, applies gradient clamping in range [-clamp, clamp] for the joint tensor only.
  optim:
    name: adamw
    lr: 1
    betas:
    - 0.9
    - 0.98
    weight_decay: 1.0e-05
    sched:
      name: NoamAnnealing
      d_model: 256
      warmup_steps: 2500
      warmup_ratio: null
      min_lr: 0.00001

trainer:
  devices: -1
  num_nodes: 1
  max_epochs: -1
  # fast_dev_run: 2000
  val_check_interval: 10000
  accelerator: auto
  strategy: ddp
  accumulate_grad_batches: 4
  gradient_clip_val: 1.0
  precision: bf16-mixed
  log_every_n_steps: 100
  enable_progress_bar: true
  num_sanity_val_steps: 5
  check_val_every_n_epoch: null
  sync_batchnorm: true
  enable_checkpointing: false
  logger: false
  benchmark: false
  max_steps: 100000
  use_distributed_sampler: false
  limit_train_batches: 200000
init_from_pretrained_model:
  model0:
    name: nvidia/stt_en_fastconformer_hybrid_medium_streaming_80ms_pc
    include:
    - encoder
```

</p>
</details> 

I'm seeing good convergence in training loss and WER but the Validation WER is way too high, 6 and 3, I was able to get 0.08 val wer on previous training runs, the only difference was the model size
After inspecting the outputs,  I found that the CTC head is producing `<unk>` token most of the time and never produces blanks, the RNNT head is the same but if I filter that token from the output, it becomes meaningful and correct, which is not the case for the ctc, below are the training and val metrics for two runs, the difference between them is the `ctc_reduction`, the yellow run is using `mean_volume` and the purple is using `mean_batch`

<img width="1863" height="918" alt="Image" src="https://github.com/user-attachments/assets/dfe19949-0b65-4ddc-9e98-e5c8b18904a5" />

**Expected behavior**

I expect similar behavior on both training and validation metrics, I understand that underfitting can happen, but I'm worried about the symptoms (no blanks and `<unk>` token produced a lot, I have tokenized the entire training dataset and not a single `<unk>` token was found
**Environment overview (please complete the following information)**
 - Method of NeMo install: pip install

**Environment details**

If NVIDIA docker image is used you don't need to specify these.
Otherwise, please provide:
- PyTorch version: 2.8
- Python version: 3.12

**Additional context**

Example: 8x H100

**Comment by MahmoudAshraf97** [comment]
I managed to solve the issue, the cause was that I applied some text transformations to the sharded manifests of a tarred datasets and reused the same tar files, supposedly they should contain the audio files only but it turns out that they still contain the same text before transformation, and hence producing `unk` token in the tokenized transcript, the solution is to recreate the tarred dataset with the transformation applied to the original manifest
